### ISSUE TITLE
Temporal Circuits: Attribution Patching to Find Time Horizon Circuits

### ISSUE THREAD

**Issue by justinshenk** [justinshenk/temporal-awareness#25]
## Overview

Extend our temporal probe findings (Layer 8, 97.1% accuracy) by identifying the *circuits* responsible for temporal encoding using mechanistic interpretability techniques.

**Key question**: Is temporal scope encoded by dedicated circuits, or distributed across the network?

## Background

Our current work shows:
- Linear probe at Layer 8 achieves 97.1% accuracy on temporal horizon classification ([probe accuracy by layer](results/temporal_tokens/probe_accuracy_by_layer.png))
- Cross-model transfer works via linear transformation (TinyLlama: +39.3%, see `results/cross_model_transfer.json`)
- Temporal encoding appears orthogonal to confidence (r=-0.017, Issue #18)

This suggests a consistent, localized representation - but we don't know *which components* write to it.

### 1. Attribution Patching
Using TransformerLens, identify which components contribute to the temporal direction:

```python
# Pseudocode
temporal_direction = probe.coef_[0]  # From trained probe at results/probes/token_temporal_layer_8_probe.pkl

# For each layer, head, MLP
for component in model.components:
    # Patch component output to zero
    patched_logits = run_with_patch(component, zero)
    
    # Measure change in temporal direction
    delta = project(patched_activations, temporal_direction)
    attribution[component] = delta
```

### 2. Activation Patching (Causal Tracing)
- Run model on short-term prompt, cache activations
- Run on long-term prompt
- Patch activations from short→long at each component
- Identify which patches flip the temporal classification

### 3. Circuit Discovery
- Identify attention heads that attend to temporal keywords
- Trace information flow from temporal tokens to residual stream
- Map the minimal circuit for temporal encoding

## Expected Outputs

1. **Attribution heatmap**: Which layers/heads/MLPs contribute most to temporal direction
2. **Minimal circuit**: Smallest set of components needed for temporal classification
3. **Cross-model comparison**: Do different architectures use similar circuits?

## Technical Requirements

- TransformerLens
- GPT-2 (primary), extend to Llama/Qwen if patterns emerge
- ~50-100 GPU hours for systematic patching

## Connection to Existing Work

- Builds on #19 (token-level scoring) - can we find circuits for token-level temporal attribution?
- Informs #20 (steering) - steering via circuit intervention vs. direction addition
- Relates to cross-model transfer findings - are circuits structurally similar across models?

## References

- Conmy et al. (2023) "Towards Automated Circuit Discovery"
- Wang et al. (2022) "Interpretability in the Wild: GPT-2 Indirect Object Identification"
- Nanda et al. "Attribution Patching" (TransformerLens tutorials)
